### PR TITLE
StreamState refactor

### PR DESCRIPTION
**Pull request by kanye-quest**
- Make stream_reader.hpp functions members of `StreamState`
- Consolidate code to update mesh and solution into `StreamState::NewMeshAndSolution`, used by threads.cpp, aux_js.cpp, and glvis.cpp
- Where possible, remove references to global `StreamState stream_state` instance

### CODE REVIEW

File: lib/stream_reader.cpp [hunk 1]
```diff
@@ -42,35 +38,31 @@ void Extrude1DMeshAndSolution(Mesh **mesh_p, GridFunction **grid_f_p,
...
       GridFunction *grid_f_2d =
-         Extrude1DGridFunction(mesh, mesh2d, *grid_f_p, 1);
+         Extrude1DGridFunction(mesh.get(), mesh2d, grid_f.get(), 1);
 
-      delete *grid_f_p;
-      *grid_f_p = grid_f_2d;
+      grid_f.reset(grid_f_2d);
    }
-   if (sol && sol->Size() == mesh->GetNV())
+   else if (sol.Size() == mesh->GetNV())
```
Comment (tomstitt) [inline lib/stream_reader.cpp:50]: Why was this changed to an `else if`? Did I copy it wrong before?

File: lib/aux_js.cpp [hunk 2]
```diff
@@ -244,64 +247,23 @@ int updateVisualization(std::string data_type, std::string stream)
...
-            vss->NewMeshAndSolution(new_m, &stream_state.sol, new_g);
-         }
-         else
-         {
-            new_g = ProjectVectorFEGridFunction(new_g);
-            VisualizationSceneVector3d *vss =
-               dynamic_cast<VisualizationSceneVector3d *>(vs);
-            vss->NewMeshAndSolution(new_m, new_g);
-         }
-      }
```
Comment (tomstitt) [inline lib/aux_js.cpp:286]: much cleaner! nice!

**Review by tzanio — APPROVED** [review]
I have not checked all the refactoring changes super-carefully, but it didn't notice any issues and the code appears to work as before.

**Review by tomstitt — APPROVED** [review]
I looked over everything briefly and the changes look good to me. I tested with glvis-js and everything I tested looks like it's working (loading files/updating with streams)
